### PR TITLE
Make dicom subdir

### PR DESCRIPTION
**Pull request by jcupitt**
This PR moves `dicom.h` into a `dicom/` subdirectory so that we can always include the libdicom headers with `<dicom/dicom.h>`.

This builds on #25 so check that one first.

### CODE REVIEW

**Comment by bgilbert** [comment]
The `pkg-config` include path should be updated to remove the `/dicom` suffix, and the `#include` in `usage.rst` will also need to be updated.  Also, this breaks the API docs build.

**Comment by jcupitt** [comment]
I fixed the pkg-config path and fixed `usage.rst`. 

I suppose the doc changes should wait until your other PR is merged.

**Comment by jcupitt** [comment]
Could someone approve this PR too?

**Review by hackermd — APPROVED** [review]

**Pull request merged**
